src/cudadev/bin/main.cc

```diff
@@ -104,11 +104,29 @@ int main(int argc, char** argv) {
   int numberOfDevices;
   auto status = cudaGetDeviceCount(&numberOfDevices);
   if (cudaSuccess != status) {
-    std::cout << "Failed to initialize the CUDA runtime";
+    std::cout << "Failed to initialize the CUDA runtime" << std::endl;
     return EXIT_FAILURE;
   }
   std::cout << "Found " << numberOfDevices << " devices" << std::endl;
 
+  for (int dev = 0; dev < numberOfDevices; ++dev) {
+    status = cudaSetDevice(dev);
+    if (cudaSuccess != status) {
+      std::cout << "Failed to set device" << std::endl;
+      return EXIT_FAILURE;
+    }
+    status = cudaSetDeviceFlags(cudaDeviceScheduleBlockingSync);
+    if (cudaSuccess != status) {
+      std::cout << "Failed to set device status flags" << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+  status = cudaSetDevice(0);
+  if (cudaSuccess != status) {
+    std::cout << "Failed to set device" << std::endl;
+    return EXIT_FAILURE;
+  }
```

Comment:
do we care to re-set the current device to 0 ?

Reply:
Nothing in our code should care it, i.e. we should always be setting the device explicitly. I'd be happy to remove this `cudaSetDevice()`. On the other hand, it might be worth to test also if with `cudaEventBlockingSync` this `cudaSetDeviceFlags(cudaDeviceScheduleBlockingSync)` has any effect, and if not, just remove all changes in this file.